typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -255,36 +255,61 @@ export class EvmERC20WarpRouteReader extends EvmRouterReader {
     return contractVerificationStatus;
   }
 
-  async getOwnerStatus(chain: ChainName, address: Address) {
-    let ownerStatus: Record<string, OwnerStatus> = {};
-    if (this.multiProvider.isLocalRpc(chain)) {
-      this.logger.debug('Skipping owner verification for local endpoints');
-      return {
-        [address]: OwnerStatus.Skipped,
-      };
-    }
-
+  /**
+   * Validates the status of an owner address directly.
+   * Assumes the address parameter is an owner address (not a contract address).
+   * @param chain - The chain name
+   * @param ownerAddress - The owner address to validate
+   * @returns Record with owner address as key and OwnerStatus as value
+   */
+  async validateOwnerAddress(
+    chain: ChainName,
+    ownerAddress: Address,
+  ): Promise<Record<string, OwnerStatus>> {
     const provider = this.multiProvider.getProvider(chain);
-    const owner = await Ownable__factory.connect(address, provider).owner();
+    const ownerStatus: Record<string, OwnerStatus> = {};
 
-    ownerStatus[owner] = (await isAddressActive(provider, owner))
+    // Check owner status (Active/Inactive/GnosisSafe)
+    ownerStatus[ownerAddress] = (await isAddressActive(provider, ownerAddress))
       ? OwnerStatus.Active
       : OwnerStatus.Inactive;
 
     // Heuristically check if the owner could be a safe by calling expected functions
     // This status will overwrite 'active' status
     try {
-      const potentialGnosisSafe = ISafe__factory.connect(owner, provider);
+      const potentialGnosisSafe = ISafe__factory.connect(
+        ownerAddress,
+        provider,
+      );
 
       await Promise.all([
         potentialGnosisSafe.getThreshold(),
         potentialGnosisSafe.nonce(),
       ]);
-      ownerStatus[owner] = OwnerStatus.GnosisSafe;
+      ownerStatus[ownerAddress] = OwnerStatus.GnosisSafe;
     } catch {
-      this.logger.debug(`${owner} may not be a safe`);
+      this.logger.debug(`${ownerAddress} may not be a safe`);
     }
 
+    return ownerStatus;
+  }
+
+  async getOwnerStatus(chain: ChainName, address: Address) {
+    let ownerStatus: Record<string, OwnerStatus> = {};
+
+    if (this.multiProvider.isLocalRpc(chain)) {
+      this.logger.debug('Skipping owner verification for local endpoints');
+      return {
+        [address]: OwnerStatus.Skipped,
+      };
+    }
+
+    const provider = this.multiProvider.getProvider(chain);
+    const owner = await Ownable__factory.connect(address, provider).owner();
+
+    // Use the extracted function to validate the owner
+    ownerStatus = await this.validateOwnerAddress(chain, owner);
```

Comment:
Should we have the `getOwnerStatus` method as a separate function so that it can be called from other places without needing to create an instance of the `EvmERC20WarpRouteReader?